x-pack/plugins/apm/server/lib/transaction_groups/get_error_rate.ts

```diff
@@ -96,30 +99,36 @@ export async function getErrorRate({
 
   const noHits = resp.hits.total.value === 0;
 
-  const erroneousTransactionsRate =
-    resp.aggregations?.total_transactions.buckets.map((bucket) => {
-      const successful =
-        bucket[EVENT_OUTCOME].buckets.find(
-          (eventOutcomeBucket) =>
-            eventOutcomeBucket.key === EventOutcome.success
-        )?.count.value ?? 0;
+  if (!resp.aggregations) {
+    return { noHits, erroneousTransactionsRate: [], average: null };
+  }
 
-      const failed =
-        bucket[EVENT_OUTCOME].buckets.find(
-          (eventOutcomeBucket) =>
-            eventOutcomeBucket.key === EventOutcome.failure
-        )?.count.value ?? 0;
+  function calculateTransactionErrorPercentage(
+    outcomeResponse: AggregationResultOf<typeof outcomes, {}>
+  ) {
+    const successfulTransactions =
+      outcomeResponse.buckets.find(
+        (bucket) => bucket.key === EventOutcome.success
+      )?.count.value ?? 0;
+    const failedTransactions =
+      outcomeResponse.buckets.find(
+        (bucket) => bucket.key === EventOutcome.failure
+      )?.count.value ?? 0;
 
+    return failedTransactions / (successfulTransactions + failedTransactions);
+  }
+
+  const erroneousTransactionsRate = resp.aggregations.timeseries.buckets.map(
+    (bucket) => {
       return {
         x: bucket.key,
-        y: failed / (successful + failed),
+        y: calculateTransactionErrorPercentage(bucket.outcomes),
       };
-    }) || [];
+    }
+  );
```

Comment:
considering this is copy/paste from `get_services_items_stats`, should we export/import a shared function?